lib/DiscoveryStreamFeed.jsm

```diff
@@ -40,6 +40,24 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
     this.cache = new PersistentCache(CACHE_KEY, true);
     // Internal in-memory cache for parsing json prefs.
     this._prefCache = {};
+    // For SPOCS Fill telemetry
+    this._spocsFill = [];
+  }
+
+  _addSpocsFill(id, reason, displayed, fullRecalc) {
+    this._spocsFill.push({
+      id,
+      reason,
+      displayed,
+      full_recalc: fullRecalc,
```

Comment:
is full recalc needed? I'm wondering, if it's not a full recalc, do we need to send any pings? Like, a spoc impression won't actually cause a page to render something, especially render something different. It's just updating the main store which isn't used for when people see. When someone goes to view a page, then we grab the data, but at that point the impressions stuff is already updated.

Reply:
According to [this comment](https://bugzilla.mozilla.org/show_bug.cgi?id=1535717#c5), I think this field is still useful for certain cases to provide more context of this ping. For example, frequency capping can be performed in either the recalculation case, or the non-recalc case.